cpp/src/arrow/util/compression_test.cc

```diff
@@ -368,6 +368,42 @@ TEST_P(CodecTest, CodecRoundtrip) {
   }
 }
 
+TEST(CodecTest, CodecRoundtripGzipMembers) {
+  std::unique_ptr<Codec> gzip_codec;
+  ASSERT_OK_AND_ASSIGN(gzip_codec, Codec::Create(Compression::GZIP));
+
+  for (int data_half_size : {0, 10000, 100000}) {
+    int64_t actual_size_p1, actual_size_p2;
+    std::vector<uint8_t> data_half = MakeRandomData(data_half_size);
+    std::vector<uint8_t> data_full(data_half.begin(), data_half.end());
+    data_full.insert(data_full.end(), data_half.begin(), data_half.end());
```

Comment:
This test is deliberately trivial:
1. the two members have the same length
2. they have the exact same content

Can you please make the test more relevant?

Reply:
You can for example generate a single piece of random data and cut it somewhere not at the half.

Reply:
done

Reply:
done